go.mod

```diff
@@ -13,7 +13,7 @@ require (
 	github.com/go-logr/logr v1.2.3
 	github.com/golang/mock v1.6.0
 	github.com/google/uuid v1.3.0
-	github.com/hashicorp/consul/api v1.15.2
+	github.com/hashicorp/consul/api v1.10.1-0.20221004184615-1b565444beb0
```

Comment:
This PR depends on functionality not yet released. Waiting for `v1.15.3` [here](https://github.com/hashicorp/consul/tags)

Reply:
This tag is expected today or tomorrow, so I'm going to go ahead and mark this PR as ready for review

Reply:
Tagged and consumed!